### PR TITLE
[CodeHealth] Drop `proc-macro-error` from `filecoin-cxx`

### PR DESCRIPTION
**Pull request by cdesouza-chromium**
This PR updates all crates around `multihash` to use the latest `multihash` version that does not depend anymore on `proc-macro-error`. This crate is being removed as it is getting flagged in security audits for being unmaintained, and it also relies on unstable rustc features.

Resolves

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: common/licenses/MIT [hunk 1]
```diff
@@ -0,0 +1,22 @@
+MIT License
+
+Copyright (c) [year] [fullname]
```
Comment (fmarier) [inline common/licenses/MIT:3]: Unfortunately, every project seems to customize this line of the MIT license and so we can't really use a common copy of it, like we can for Apache-2.0 or MPL-2.0.

Did you actually find a project/library that had a copyright line with these placeholders? (I tried looking through the PR but the GitHub UI is unusable given the size)

Comment (cdesouza-chromium) [reply common/licenses/MIT:3]: This is an example https://github.com/multiformats/rust-cid. There is no license file, and MIT is mentioned by passing at the bottom of the README.md, but the link to it is broken.

Comment (cdesouza-chromium) [reply common/licenses/MIT:3]: Actually I've managed to find the actual license for these crates, added it to the crates.

File: components/filecoin/rs/Cargo.toml [hunk 2]
```diff
@@ -9,16 +9,16 @@ license = "MPL-2.0"
 base64 = "0.13.0"
 bls-signatures = { version = "0.12", default-features = false, features = ["pairing"] }
 cxx = "1"
-fvm_shared = { version = "3.10.0" }
+fvm_shared = { version = "^4.5.1" }
```
Comment (cypt4) [inline components/filecoin/rs/Cargo.toml:12]: Why ^ qualifier is used here?

File: third_party/rust/chromium_crates_io/Cargo.toml [hunk 3]
```diff
@@ -26,6 +26,7 @@ adblock-cxx = { version = "1" }
 constellation-cxx = "0.1"
 challenge-bypass-ristretto-cxx = "1"
 zcash = "1"
+subtle = "2.6.1"
```
Comment (cypt4) [inline third_party/rust/chromium_crates_io/Cargo.toml:29]: Why do we need direct dependency on subtle here?

File: components/filecoin/rs/Cargo.toml (hunk 2 above)
Comment (cdesouza-chromium) [reply components/filecoin/rs/Cargo.toml]: We don't. Fixed. I've rearranged the changes so the `gnrt` mechanical changes are in a separate commit from the ones done by me.

File: third_party/rust/chromium_crates_io/Cargo.toml (hunk 3 above)
Comment (cdesouza-chromium) [reply third_party/rust/chromium_crates_io/Cargo.toml]: We don't. Removed.

**Review by antonok-edm — APPROVED** [review]

File: third_party/rust/chromium_crates_io/Cargo.lock [hunk 4]
```diff
@@ -354,7 +354,7 @@ version = "2.0.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 dependencies = [
  "base64",
- "crypto-mac 0.11.1",
+ "crypto-mac 0.11.0",
```
Comment (bridiver) [inline third_party/rust/chromium_crates_io/Cargo.lock:357]: these went to a lower version? That's odd

**Review by bridiver — APPROVED** [review]

**Pull request merged**